src/main/java/pl/javastart/streamsexercise/PaymentService.java

```diff
@@ -19,91 +20,138 @@ class PaymentService {
     Znajdź i zwróć płatności posortowane po dacie rosnąco
      */
     List<Payment> findPaymentsSortedByDateAsc() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll()
+                .stream()
+                .sorted(Comparator.comparing(Payment::getPaymentDate))
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności posortowane po dacie malejąco
      */
     List<Payment> findPaymentsSortedByDateDesc() {
-        throw new RuntimeException("Not implemented");
+        List<Payment> paymentsSortedByDateDesc = findPaymentsSortedByDateAsc();
+        Collections.reverse(paymentsSortedByDateDesc);
+        return paymentsSortedByDateDesc;
     }
 
     /*
     Znajdź i zwróć płatności posortowane po liczbie elementów rosnąco
      */
     List<Payment> findPaymentsSortedByItemCountAsc() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll()
+                .stream()
+                .sorted(Comparator.comparingInt(p -> p.getPaymentItems().size()))
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności posortowane po liczbie elementów malejąco
      */
     List<Payment> findPaymentsSortedByItemCountDesc() {
-        throw new RuntimeException("Not implemented");
+        List<Payment> paymentsSortedByItemCountDesc = findPaymentsSortedByItemCountAsc();
+        Collections.reverse(paymentsSortedByItemCountDesc);
+        return paymentsSortedByItemCountDesc;
     }
 
     /*
     Znajdź i zwróć płatności dla wskazanego miesiąca
      */
     List<Payment> findPaymentsForGivenMonth(YearMonth yearMonth) {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll()
+                .stream()
+                .filter(p -> yearMonth.getMonth() == p.getPaymentDate().getMonth())
```

Comment:
sprawdzasz tylko miesiąc, brakuje powrównania roku. `YearMonth` to reprezentacja np kwietnia 2021, a nie tylko kwietnia